pkg/storage/ingest/util.go

```diff
@@ -171,14 +171,24 @@ func setDefaultNumberOfPartitionsForAutocreatedTopics(cfg KafkaConfig, logger lo
 	defer adm.Close()
 
 	defaultNumberOfPartitions := fmt.Sprintf("%d", cfg.AutoCreateTopicDefaultPartitions)
-	_, err = adm.AlterBrokerConfigsState(context.Background(), []kadm.AlterConfig{
+	responses, err := adm.AlterBrokerConfigsState(context.Background(), []kadm.AlterConfig{
 		{
 			Op:    kadm.SetConfig,
 			Name:  "num.partitions",
 			Value: &defaultNumberOfPartitions,
 		},
 	})
 
+	// Check if any error has been returned as part of the response.
+	if err == nil {
+		for _, res := range responses {
+			if res.Err != nil {
+				err = res.Err
+				break
```

Comment:
We're here looking at the sub-errors. What do you think about recording _all_ non-nil sub-errors into a multierr or something like that, rather than just the first?

Reply:
We issue an operation on a single resource, there shouldn't be more than 1 error.